sdks/go/pkg/beam/core/runtime/pipelinex/replace_test.go

```diff
@@ -310,6 +311,28 @@ func TestComputeInputOutput(t *testing.T) {
 	}
 }
 
+func BenchmarkComputeInputOutput(b *testing.B) {
+	in := make(map[string]*pipepb.PTransform)
+	for i := 0; i < 3000; i++ {
```

Comment:
Can we add a comment about what this graph looks like? I think it's just a linked list of nested primitives and composites, but I'd like a quick confirmation.